frontend/src/App.tsx

```diff
@@ -182,6 +184,8 @@ export class App extends PureComponent<Props, State> {
 
   private readonly componentRegistry: ComponentRegistry
 
+  private readonly appId: string = generateUID()
```

Comment:
nit: I'd name this something like embeddingId because appId has the potential to conflict with names used for other reasons unrelated to embedding